diff -r c294faf5fce5 src/video/cocoa/SDL_cocoaopengl.m
--- a/src/video/cocoa/SDL_cocoaopengl.m	Wed Aug 15 21:00:33 2012 -0400
+++ b/src/video/cocoa/SDL_cocoaopengl.m	Sat Aug 25 19:48:49 2012 -0700
@@ -81,16 +81,23 @@ Cocoa_GL_CreateContext(_THIS, SDL_Window
     pool = [[NSAutoreleasePool alloc] init];
 
 #ifndef FULLSCREEN_TOGGLEABLE
     if (window->flags & SDL_WINDOW_FULLSCREEN) {
         attr[i++] = NSOpenGLPFAFullScreen;
     }
 #endif
 
+    if(_this->gl_config.major_version == 3 && 
+       _this->gl_config.minor_version == 2)
+    {
+        attr[i++] = NSOpenGLPFAOpenGLProfile;
+        attr[i++] = NSOpenGLProfileVersion3_2Core; 
+    }
+
     attr[i++] = NSOpenGLPFAColorSize;
     attr[i++] = SDL_BYTESPERPIXEL(display->current_mode.format)*8;
 
     attr[i++] = NSOpenGLPFADepthSize;
     attr[i++] = _this->gl_config.depth_size;
 
     if (_this->gl_config.double_buffer) {
         attr[i++] = NSOpenGLPFADoubleBuffer;
